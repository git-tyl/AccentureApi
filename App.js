import React, { Component } from "react";
// import QuestionList from "./Components/QuestionList"
import UserList from "./Components/UserList"
// export default QuestionList

import {
  StackNavigator,
} from 'react-navigation';

const RootStack = StackNavigator({
  Home: { screen: UserList },
});


export default class App extends React.Component {
  render() {
    return <RootStack />;
  }
}
