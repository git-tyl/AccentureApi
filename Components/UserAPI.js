import axios from 'axios';

const API_STEM = "https://private-fd094-accentureapi1.apiary-mock.com";
const LIST_NAME = "users"


//takes in a list name that gives a list of the type of group
//returns a response object with status , JSON etc..
function fetchUsers(list_name = LIST_NAME) {
  let url = `${API_STEM}/${LIST_NAME}`;

  // return fetch(url)
  //   .then(response =>
  //     response.json()
  //   )
  //   .then(responseJson => {
  //     return responseJson;
  //   })
  //   .catch(error => {
  //     console.error(error);
  //   });

  return axios.get(url)
    .then(response => {
      // console.log(response)
        return response
      }
    )
    .catch(error => {
      console.error(error);
    });
}

function postUsers(userName,userTitle) {

  let url = `${API_STEM}/${LIST_NAME}`;
  fetch(url, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      name: userName,
      title: userTitle,
    }),
  })
  .then(response => response.json())
  .then(responseJson => {
    return responseJson;
  }).catch(error => {
    console.error(error);
  });

}

//gets one single user
function findUser(userID) {

  let url = `${API_STEM}/${LIST_NAME}/${userID}`;
  return fetch(url)
    .then(response => response.json())
    .then(responseJson => {
      return responseJson;
    })
    .catch(error => {
      console.error(error);
    });

}

function updatesUser(userID) {

  let url = `${API_STEM}/${LIST_NAME}/${userID}`;
  return fetch(url)
    .then(response => response.json())
    .then(responseJson => {
      return responseJson;
    })
    .catch(error => {
      console.error(error);
    });

}


function deleteUser(id) {

  let url = `${API_STEM}/${LIST_NAME}/${id}`;
  fetch(url, {
    method: 'DELETE',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    }
  })
  .then(response => response.json())
  .then(responseJson => {
    return responseJson;
  })
  .catch(error => {
    console.error(error);
  });

}

export default { fetchUsers: fetchUsers,postUsers:postUsers };
