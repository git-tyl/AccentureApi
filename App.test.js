import React from 'react';
import App from './App';
import UserAPI from "./Components/UserAPI"
import renderer from 'react-test-renderer';

//the problem
//https://github.com/facebook/jest/issues/2071

// const XMLHttpRequest = require('xhr2');
// global.XMLHttpRequest = XMLHttpRequest;
// require("isomorphic-fetch");

it('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});

//REST TESTS

describe('Get All Users',() => {

  let response;

  beforeAll(async () => {
    response = await UserAPI.fetchUsers();
  });

  test('gets all users return 200', () => {
    // let response = await UserAPI.fetchUsers();
    expect(response.status).toBe(200);
  });

  test('gets all users returns correct data', () => {

    console.log(response.data.results);
    expect(response.data.results).toMatchSnapshot();
  });
});

//POST test positive
test('post test create user', async () => {
  // const response = await API.login('test-account@example.com', 'supersecret!');
  //     expect(response).toMatchSnapshot();
});

//POST test negative
test('post test create user', async () => {

});

//PUT test positive
test('able to update user object', async () => {

});

//PUT test negative
test('unable to update user object due to wrong data', async () => {

});

//Delete Test positive
test('able to deletes user', async () => {

});

//Delete Test negative
test('unable to deletes user due to wrong data', async () => {

});
