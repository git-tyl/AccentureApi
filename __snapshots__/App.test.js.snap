// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get All Users gets all users returns correct data 1`] = `
Array [
  Object {
    "id": 1,
    "name": "Jay",
    "title": "Manager",
  },
  Object {
    "id": 2,
    "name": "Jill",
    "title": "Employee",
  },
  Object {
    "id": 3,
    "name": "Sam",
    "title": "Employee",
  },
  Object {
    "id": 4,
    "name": "Jane",
    "title": "Employee",
  },
]
`;
